javalib/src/main/scala/java/lang/process/UnixFileDescriptorAtomic.scala

```diff
@@ -0,0 +1,32 @@
+package java.lang.process
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import scala.scalanative.annotation.alwaysinline
+import scala.scalanative.posix.unistd
+
+class UnixFileDescriptorAtomic(val value: AtomicInteger) extends AnyVal {
+
+  @alwaysinline def get(): Int = value.get()
+
+  @alwaysinline def release(): Int = value.getAndSet(-1)
+
+  @alwaysinline def close(): Unit = {
+    val prev = release()
+    if (prev != -1) unistd.close(prev)
+  }
+
+}
+
+object UnixFileDescriptorAtomic {
+
+  @alwaysinline
+  def apply(value: AtomicInteger): UnixFileDescriptorAtomic =
+    new UnixFileDescriptorAtomic(value)
```

Comment:
This one seems to be redundant, we can expose just a single apply method taking Int, allowing us to hide implementation detail, but it's not that much important